### DISCUSSION TITLE
Can we disable an annotation at runtime?

### DISCUSSION THREAD

**Discussion by moghaddam** [eclipse/microprofile-fault-tolerance#625] [Q&A] [answered]
Hi

As part of adding Fault Tolerance to our existing project, I want to know if I can disable them at runtime in case something goes wrong in the early releases. I tried to set the following at runtime in `System.properties` and seems ineffective:

```
Timeout/enabled=false
```
When I restart the system, then it gets disabled. So, apparently configurations are being read only once in system startup and changing them at runtime has no effect. But I could not find this mentioned in documentation explicitly. So I just wanted to be sure my interpretation is correct and also suggest to add this minor detail to the spec to avoid misinterpretation.

Regards
Ehsan

**Comment by Emily-Jiang** [comment, marked as answer]
Yes, all of the configuration are static and only read once at the application starting to avoid inconsistent behaviour.

**Comment by Emily-Jiang** [comment]
I checked the spec, which only states one of the configurations being read only. The spec would need to state that the same rule is applicable to all configurations.

> In order to prevent from any unexpected behaviours, the property MP_Fault_Tolerance_NonFallback_Enabled will only be read on application starting. Any dynamic changes afterwards will be ignored until the application restarting.

**Reply by moghaddam** [reply]
Thanks Emily for the quick answer :slightly_smiling_face:
